### ISSUE TITLE
Generate the DTO's in separate file

### ISSUE THREAD

**Issue by andrewinkler1** [RSuter/NSwag#1009]
Is it possible to generate the dto's outside the service methods in a separate file?

**Comment by RSuter** [comment]
Currently only for C#, 

Duplicate: https://github.com/RSuter/NJsonSchema/issues/514

**Comment by RSuter** [comment]
You could however exclude all DTOs from generation (ExcludedTypeNames) generate, then exclude all clients generate again...

**Comment by RSuter** [comment]
Duplicate: https://github.com/RSuter/NSwag/issues/294

**Comment by statler** [comment]
Something like this will do the job using NJsonSchema.CodeGeneration.TypeScript. Note that you will need to change your ts template to deal with your imports

`
	using System;
	using System.Collections.Generic;
	using System.IO;
	using System.Windows.Forms;
	using System.Text.RegularExpressions;
	using NSwag;
	using NJsonSchema.CodeGeneration.TypeScript;
	using NJsonSchema.CodeGeneration;

	namespace TypescriptFileSplitter
	{
		public partial class getTs
		{
			public async void getTS(object sender, EventArgs e)
			{
				if (!Directory.Exists(beDest.Text))
				{
					MessageBox.Show("The target directory does not exist.");
					return;
				}

				var schemaSwagger = await SwaggerDocument.FromUrlAsync("http://localhost:60000/swagger/v2/swagger.json");

				TypeScriptGeneratorSettings tsGenSettings = new TypeScriptGeneratorSettings()
				{
					ConvertConstructorInterfaceData = true,
					TypeStyle = TypeScriptTypeStyle.KnockoutClass,
					MarkOptionalProperties = false,
					PropertyNameGenerator = new myPropertyNameGenerator(),
					TypeNameGenerator = new myTypeNameGenerator(),
					TypeScriptVersion = 2.4m
				};

				var _resolver = new TypeScriptTypeResolver(tsGenSettings);
				_resolver.RegisterSchemaDefinitions(schemaSwagger.Definitions);
				var generatorJ = new TypeScriptGenerator(schemaSwagger, tsGenSettings, _resolver);
				var typeDef = generatorJ.GenerateTypes();
				foreach (CodeArtifact codeArtifact in typeDef.Artifacts)
				{
					File.WriteAllText(Path.Combine(beDest.Text, codeArtifact.TypeName + ".ts"), codeArtifact.Code);
				}
			}
		}
	}
`

**Comment by schuettecarsten** [comment]
Any news on this?

**Comment by jeremy-collette** [comment]
@RicoSuter which option do we enable to have the DTOs in a separate file for C#? I just see an option for one output file. I'm on v12.3.1.

**Comment by RicoSuter** [comment]
This option is not yet available

**Comment by IvanJosipovic** [comment]
+1 needing this.

**Comment by joni1700** [comment]
+1

**Comment by kimlundjohansen** [comment]
+1 need this

**Comment by justin-c-wong** [comment]
+1

**Comment by RicoSuter** [comment]
For C# output you can already specify a clients and separate contracts file output.

**Comment by milanjaros** [comment]
To generate contracts in separate file use `/GenerateContractsOutput:true /ContractsOutput:path/to/Dtos.Generated.cs /ContractsNamespace:Namespace.To.Dtos.Generated`
You could also skip generating client by `/GenerateClientClasses:false`, so you can also ommit `/output:...`

**Comment by justin-c-wong** [comment]
Any plans to support separation of DTOs generated in Typescript?

**Comment by milanjaros** [comment]
@justin-c-wong, do you mean to have every class and interface in separate files? Is that what you really want for generated code?
Or do you mean separation of DTOs from Client? ... Which can be achieved by these arguments:
`/output:path/to/Dtos.Generated.ts /namespace:Path.To.Dtos.Generated /GenerateContractsOutput:true /GenerateClientClasses:false`
Note, there are not `/ContractsOutput` and `/ContractsNamespace` as in example abov.

**Comment by lafritay** [comment]
@milanjaros I'm able to get the clients and types generated in separate files using the method you mention above. I'm not able to figure out how to get the custom types in the Clients file to reference the types in the Types file though. Any tips on how to make that happen?

**Comment by jeremyVignelles** [comment]
@lafritay : Generate them in the same namespace? If you do it that way and use their class name, the compiler should be able to find them without `using`

**Comment by lafritay** [comment]
@jeremyVignelles Sorry, I meant for typescript, not C#. Unless the namespacing feature was supposed to work for typescript as well? It doesn't seem to work for me...

**Comment by jeremyVignelles** [comment]
Indeed, sorry. I don't know if you can add your custom imports in the generated file by configuration, but you could create your own file template and put your using before the original code
